tests.py

```diff
@@ -1,41 +1,62 @@
+import json
+from mock import patch
+import collections
+from datetime import datetime
 from cStringIO import StringIO
 from txwrapplog import wrapp_observer, Logger
 
 
 
 
 class TestWrappObserver(object):
-    def setup(self):
+    @patch('txwrapplog._timestamp')
+    def setup(self, timestamp_mock):
+        self.timestamp = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+        timestamp_mock.return_value = self.timestamp
+        self.msg = 'Hello'
+        self.service = "api"
+        self.host = "host-01"
         self.out = StringIO()
-        self.log = Logger(observer=wrapp_observer(self.out))
+        self.log = Logger(observer=wrapp_observer(self.out, service=self.service, host=self.host))
+
+    def _generate_output(self, level):
```

Comment:
I would expand the definition of _generate_output to accept the message and level. I would be happy if we remove the internal magic of fetching the message and passing the information from the function call itself. What do you think ?